wordpress-mobile/WordPress-iOS
# Create site creation error views and integrate with final phase of site creation

### Description

Addresses the first phase of #10429.

**NB : This PR does not impact release notes!**

### Testing

- Checkout the branch and confirm that both new & old tests pass.
- Apply a change to account for [this fix](https://github.com/wordpress-mobile/WordPress-iOS/pull/10690).
- Navigate through the entire _Site Creation_ flow to exercise the success flow and confirm that no regressions have been encountered.
- Revert [`67a53a5`](https://github.com/wordpress-mobile/WordPress-iOS/commit/67a53a5e9d7c791d6657e5b5f53d3b2647edae5a) to toggle the mock to fail and repeat the preceding step.
- Visually confirm that each of the three actions (i.e., retry, contact support, dismiss) exposed behave as expected.

### Working Software

The GIF below shows an iPhone XS Max exercising each of the failure cases related to site creation.

![10492_pr](https://user-images.githubusercontent.com/221062/50256587-cd4a4b80-03ab-11e9-9f8e-579105b77778.gif)

### Visual QA

@SylvesterWilmott : in addition to the animated GIF above, I wanted to share the following screenshots for your review. They represent each of the error views conceived for the project on our smallest & largest canvases.

All of the Zeplin specifications were honored, save for the metrics related to the dismiss button. For that, I expressed the horizontal & vertical insets as a percentage of screen size. Please advise if you would like me to revise this further.

#### iPhone SE (Site Creation, General, Network Unreachable)

_Site Loading_
<img width="504" alt="10429_static-error_siteloading_iphone" src="https://user-images.githubusercontent.com/221062/50256623-fe2a8080-03ab-11e9-877e-f1e07b0e061c.png">

_General_
<img width="504" alt="10429_static-error_general_iphone" src="https://user-images.githubusercontent.com/221062/50256643-139faa80-03ac-11e9-8987-d05ea240f4a1.png">

_Network Unreachable_
<img width="504" alt="10429_static-error_networkunreachable_iphone" src="https://user-images.githubusercontent.com/221062/50256625-fec31700-03ab-11e9-839f-cbaceba4f5b1.png">

#### iPad Pro (Portrait)

_Site Loading_
<img width="878" alt="10429_static-error_siteloading_ipad-portrait" src="https://user-images.githubusercontent.com/221062/50256678-40ec5880-03ac-11e9-81d8-480dfb52b7df.png">

_General_

<img width="878" alt="10429_static-error_general_ipad-portait" src="https://user-images.githubusercontent.com/221062/50256624-fe2a8080-03ab-11e9-82de-cb1dc2782d8d.png">

_Network Unreachable_
<img width="834" alt="10429_static-error_networkunreachable_ipad-portait" src="https://user-images.githubusercontent.com/221062/50256685-4c3f8400-03ac-11e9-9881-16aaf28b4bd0.png">

#### iPad Pro (Landscape)

_Site Loading_
<img width="1109" alt="10429_static-error_siteloading_ipad-landscape" src="https://user-images.githubusercontent.com/221062/50256694-53ff2880-03ac-11e9-95c9-985c9c5437ac.png">

_General_
<img width="1109" alt="10429_static-error_general_ipad-landscape" src="https://user-images.githubusercontent.com/221062/50256700-59f50980-03ac-11e9-83f6-c73a37300dd8.png">

_Network Unreachable_
<img width="1109" alt="10429_static-error_networkunreachable_ipad-landscape" src="https://user-images.githubusercontent.com/221062/50256703-5eb9bd80-03ac-11e9-8d7e-5b6b9e794b98.png">
